ananse/influence.py

```diff
@@ -33,41 +33,165 @@
+            )
+            activity = 0 if len(row) < edge_atribute_length else float(row["activity"])
+            try:
+                G.add_edge(
+                    source,
+                    target,
+                    weight=weight,
+                    weighted_binding=weighted_binding,
+                    TF_expression=tf_expression,
+                    TF_activity=activity,
+                    TG_expression=target_expression,
+                )
+            except Exception:
+                print("Could not parse edge weight.")
+                raise
+
+        # if there is only prob score, only that is inc in the GRN
+        else:
+            try:
+                G.add_edge(source, target, weight=weight, n=1)
+            except Exception:
+                print("Could not parse edge weight.")
+                raise
     return G
 
 
-def difference(S, R):
-    """Calculate the network different between two cell types."""
-    DIF = nx.create_empty_copy(R)
-    for (u, v, d) in S.edges(data=True):
-        if (u, v) not in R.edges:
-            DIF.add_edge(u, v, weight=d["weight"], n=1)
-        else:
-            diff_weight = S.edges[u, v]["weight"] - R.edges[u, v]["weight"]
-            if diff_weight > 0:
+def difference(GRN_source, GRN_target):
+    """Calculate the network different between two GRNs.
+    It first takes the nodes from both networks, and subsequently
+    First adds the edges from the target network that are missing in
+    the source network.
+    Secondly add the edges present in both but with a higher interaction
+    score in the target network
+    """
+    DIF = nx.create_empty_copy(GRN_source)  # copying source GRN nodes
+    nodes_target = nx.create_empty_copy(GRN_target)  # copying target GRN nodes
+    DIF.add_nodes_from(
+        list(nodes_target.nodes)
+    )  # merge the nodes if there are differences
+    # (which can happen) when taking the header instead of the --unnion-interactions flagg
+
+    # lets check if the full GRN output is loaded and if so output all atributes to the diffnetwork:
+    full_output_GRN = bool(nx.get_edge_attributes(GRN_source, "TF_expression"))
+    if full_output_GRN:
+        logger.info("calculating diff GRN with full output")
+        # lets load all the differential edges into the diffnetwork
+        for (u, v, a) in GRN_target.edges(data=True):
+            if (u, v) not in GRN_source.edges:
+                # if the edge is not in the target network (when using head instead of union of interactions)
+                # , add all source atributes to the diff network and put the target atributes to 0
                 DIF.add_edge(
-                    u, v, weight=diff_weight, n=1, neglogweight=-np.log(diff_weight)
+                    u,
+                    v,
+                    weight=a["weight"],
+                    n=1,
+                    source_weight="NA",
```

Comment:
no idea about this, but shouldn't you add `None` instead of a string, or maybe `np.NaN`?

Reply:
I'll just remove this part, when we use the intersect of GRNS when multiple networks are suplied, there shouldnt be any vallues missing :)